api-report/sequence.api.md

```diff
@@ -314,8 +320,20 @@ export enum IntervalType {
 // @public (undocumented)
 export interface IOverlappingIntervalsIndex<TInterval extends ISerializableInterval> extends IntervalIndex<TInterval> {
     // (undocumented)
-    findOverlappingIntervals(start: number, end: number): TInterval[];
-    gatherIterationResults(results: TInterval[], iteratesForward: boolean, start?: number, end?: number): void;
+    findOverlappingIntervals(start: number | {
```

Comment:
The `{ pos: number; side: Side }` type will be common in interval collection APIs. we should name it (maybe "SequencePlace" or something similar) and use it.

This also has the benefit of making its semantics documentable in a standalone fashion, e.g. some of what you wrote on Side kind of belongs there instead :)